src/regression_tests/utils.ts

```diff
@@ -0,0 +1,183 @@
+import { SpvWalletClient } from '../client';
+import { generateKeys } from '../utils/keys';
+import { TransactionFilter } from '../filters';
+import { QueryParams, Recipients, Tx, Txs } from '../types';
+
+const CLIENT_ONE_URL_ENV_VAR = 'CLIENT_ONE_URL';
+const CLIENT_TWO_URL_ENV_VAR = 'CLIENT_TWO_URL';
+const CLIENT_ONE_LEADER_XPRIV_ENV_VAR = 'CLIENT_ONE_LEADER_XPRIV';
+const CLIENT_TWO_LEADER_XPRIV_ENV_VAR = 'CLIENT_TWO_LEADER_XPRIV';
+const AT_SIGN = '@';
+const DOMAIN_PREFIX = 'https://';
+const API_VERSION = 'v1';
+const ERR_EMPTY_XPRIV_ENV_VARIABLES = 'missing xpriv variables';
+const EXPLICIT_HTTP_URL_REGEX = RegExp('^https?://');
+
+export interface RegressionTestUser {
+  xpriv: string;
+  xpub: string;
+  paymail: string;
+}
+
+export interface RegressionTestConfig {
+  clientOneURL: string;
+  clientTwoURL: string;
+  clientOneLeaderXPriv: string;
+  clientTwoLeaderXPriv: string;
+}
+
+// getEnvVariables retrieves the environment variables needed for the regression tests.
+export const getEnvVariables = function (): RegressionTestConfig {
+  const rtConfig: RegressionTestConfig = {
+    clientOneURL: process.env[CLIENT_ONE_URL_ENV_VAR] || '',
+    clientTwoURL: process.env[CLIENT_TWO_URL_ENV_VAR] || '',
+    clientOneLeaderXPriv: process.env[CLIENT_ONE_LEADER_XPRIV_ENV_VAR] || '',
+    clientTwoLeaderXPriv: process.env[CLIENT_TWO_LEADER_XPRIV_ENV_VAR] || '',
+  };
+
+  if (rtConfig.clientOneLeaderXPriv === '' || rtConfig.clientTwoLeaderXPriv === '') {
+    throw new Error(ERR_EMPTY_XPRIV_ENV_VARIABLES);
+  }
+
+  if (rtConfig.clientOneURL === '' || rtConfig.clientTwoURL === '') {
+    rtConfig.clientOneURL = 'http://localhost:3003/v1';
+    rtConfig.clientTwoURL = 'http://localhost:3003/v1';
+  }
+
+  rtConfig.clientOneURL = addPrefixIfNeeded(rtConfig.clientOneURL);
+  rtConfig.clientTwoURL = addPrefixIfNeeded(rtConfig.clientTwoURL);
+
+  rtConfig.clientOneURL = addApiVersionIfNeeded(rtConfig.clientOneURL, API_VERSION);
+  rtConfig.clientTwoURL = addApiVersionIfNeeded(rtConfig.clientTwoURL, API_VERSION);
+
+  return rtConfig;
+};
+
+// getPaymailDomain retrieves the shared configuration from the SPV Wallet.
+export const getPaymailDomain = async function (xpriv: string, clientUrl: string): Promise<string> {
+  const wc = new SpvWalletClient(clientUrl, {
+    xPriv: xpriv,
+  });
+
+  const sharedConfig = await wc.GetSharedConfig();
+
+  if (sharedConfig.paymail_domains.length != 1) {
+    throw new Error(`expected 1 paymail domain, got ${sharedConfig.paymail_domains.length}`);
+  }
+
+  return sharedConfig.paymail_domains[0];
+};
+
+// createUser creates a set of keys and new paymail in the SPV wallet
+export const createUser = async function (
+  paymail: string,
+  paymailDomain: string,
+  instanceUrl: string,
+  adminXPriv: string,
+): Promise<RegressionTestUser> {
+  const keys = generateKeys();
```

Comment:
just a thought:
isn't it worth to wrap this business logic in `try{} catch()?`

Not a blocker at all, just thinking about it (GO doesn't have this feature, so it's omitted there).

Reply:
I thought about it, but since these are tests what would you suggest we do in a catch? I didn't see the point in trycatching here since if it throws, test will fail either way. We could either just clog the error and rethrow it or we could catch and throw different error with different message. Do you think there would be any benefit in catching and throwing different message?

Reply:
We can ommit it for now.